lib/rulezilla/dsl.rb

```diff
@@ -84,8 +84,13 @@ def record_klass_instance(record)
       end
 
       def missing_attributes(record)
-        record = OpenStruct.new(record) if record.is_a?(Hash)
-        mandatory_attributes.map(&:to_sym) - record.methods
+        record_attributes = if [Hash, OpenStruct].include? record.class
```

Comment:
I was going to suggest using `is_a?` meaning you have something a little more fluent:
```
if record.is_a?(Hash) || record.is_a?(OpenStruct)
```